src/script/chem/molfile/molfile.js

```diff
@@ -286,9 +286,135 @@ Molfile.prototype.writeCTab3000 = function (molecule) { // eslint-disable-line m
 	this.writeCR(`M  V30 COUNTS ${noOfAtoms} ${noOfBonds} ${noOfSGroups} 0 ${isChiral}`);
 	this.writeAtomBlock3000(molecule);
 	this.writeBondBlock3000(molecule);
+	this.writeSGroupBlock3000(molecule);
 	this.writeCR('M  V30 END CTAB');
 };
 
+Molfile.prototype.writeSGroupBlock3000 = function (molecule) {
+	if (molecule.sgroups.size === 0) {
+		return;
+	}
+	this.writeCR('M  V30 BEGIN SGROUP');
+	var sgmap = {};
+	var cnt = 1;
+	var sgmapback = {};
+	var sgorder = molecule.sGroupForest.getSGroupsBFS();
```

Comment:
could you make these variable names a little more descriptive and camelCase?

Reply:
Aah actually this did seem cryptic to me as well :(

I should have mentioned, slipped from my head somehow - I have plucked the "algorithm" to iterate over the S-Groups in the molecule from the V2000 writer, in this file itself -https://github.com/schrodinger/ketcher/blob/master/src/script/chem/molfile/molfile.js#L630

Agreed that the variable naming isn't that great and gives little information.
But I'll try my best renaming them here, with what best I can deduce